### DISCUSSION TITLE
How to make python 3(.7) default for CodeQL

### DISCUSSION THREAD

**Discussion by RhinosF1** [community/community#25008] [Actions]
<p>Hi,</p>
<p>Does anyone know how to make the CodeQL Code scanning alerts action by default run python 3, preferably 3.7, by default and not install dependencies using python2?</p>
<p>See <a href="https://github.com/MirahezeBots/MirahezeBots/pull/319/checks?check_run_id=1446587420#step:3:64" rel="noopener nofollow ugc">https://github.com/MirahezeBots/MirahezeBots/pull/319/checks?check_run_id=1446587420#step:3:64</a></p>
<p>It’s kind of ironic our CI is testing on a version we don’t support!!!</p>

**Comment by Daverlo** [comment, marked as answer]
<p>Hi!</p>
<p>The link you have provided points to the log of installing pip for python 2, not the install of the dependencies. At that point in time, the CodeQL action doesn’t know the python version, so it installs the tools for both python 2 and python 3.</p>
<p>When the action tries to guess the python version it defaults to python 3 <a href="https://github.com/MirahezeBots/MirahezeBots/pull/319/checks?check_run_id=1446587420#step:3:239" rel="noopener nofollow ugc">https://github.com/MirahezeBots/MirahezeBots/pull/319/checks?check_run_id=1446587420#step:3:239</a></p>
<p>By default the python 3 version available in the runners is 3.6.9, but you can change it by using the <a href="https://github.com/actions/setup-python" rel="noopener nofollow ugc">actions/setup-python</a></p>

**Comment by RhinosF1** [comment]
<p>I’ve fixed it. Thanks for the advice.</p>
